app/adapters/request-config.js

```diff
@@ -8,4 +9,20 @@ export default V3Adapter.extend({
     const url = `${apiEndpoint}/repo/${id}/request/config`;
     return this.ajax(url, 'POST', { data });
   },
+
+  handleResponse(status, headers, body) {
+    if (status == 400) {
+      let error = JSON.parse(body.error_message).error;
+      let type = error.type;
+      let message = error.message; // TODO API should unwrap this
+
+      // this is really shoehorning our custom error type and message
+      // through Ember's limited AdapterError API. would it be better
+      // to define a custom error class and overwrite its constructor
+      // or something?
```

Comment:
no, i was hoping for you-all's advice on this. i think Ember's api for errors is very restricted there, but still we do want to use an `AdapterError` (because we use the store for submitting these requests) ... what we have here now does work, but it's most definitely not a nice solution.

Reply:
i have added that question to a new section "to be discussed/resolved" on https://github.com/travis-ci/travis-web/pull/2399 in order to keep track of this